safe-report-comments.php

```diff
@@ -261,132 +275,149 @@ public function mark_comment_moderated( $comment ) {
-			$transient = get_transient( md5( $this->_storagecookie . $_SERVER['REMOTE_ADDR'] ) );
+			$transient = get_transient( md5( $this->_storagecookie . $remote_addr ) );
 			if ( !$transient ) {
-				set_transient( md5( $this->_storagecookie . $_SERVER['REMOTE_ADDR'] ), array( $comment_id => 1), $this->transient_lifetime );
+				set_transient( md5( $this->_storagecookie . $remote_addr ), array( $comment_id => 1), $this->transient_lifetime );
 			} else {
-				if ( ! isset( $transient[ $comment_id ] ) )
+				if ( ! isset( $transient[ $comment_id ] ) ) {
 					$transient[ $comment_id ] = 0;
+				}
 				$transient[ $comment_id ]++;
-				set_transient( md5( $this->_storagecookie . $_SERVER['REMOTE_ADDR'] ), $transient, $this->transient_lifetime );
+				set_transient( md5( $this->_storagecookie . $remote_addr ), $transient, $this->transient_lifetime );
 			}
 
-				
+
 			$threshold = (int) get_option( $this->_plugin_prefix . '_threshold' );
 			$current_reports = get_comment_meta( $comment_id, $this->_plugin_prefix . '_reported', true );
 			$current_reports++;
 			update_comment_meta( $comment_id, $this->_plugin_prefix . '_reported', $current_reports );
-			
-			
+
+
 			// we will not flag a comment twice. the moderator is the boss here.
 			$already_reported = get_comment_meta( $comment_id, $this->_plugin_prefix . '_reported', true );
 			$already_moderated = get_comment_meta( $comment_id, $this->_plugin_prefix . '_moderated', true );
 			if ( true == $already_reported && true == $already_moderated ) {
 				// But maybe the boss wants to allow comments to be reflagged
-				if ( ! apply_filters( 'safe_report_comments_allow_moderated_to_be_reflagged', false ) ) 
+				if ( ! apply_filters( 'safe_report_comments_allow_moderated_to_be_reflagged', false ) ) {
 					return;
+				}
 			}
 
 			if ( $current_reports >= $threshold ) {
 				do_action( 'safe_report_comments_mark_flagged', $comment_id );
 				wp_set_comment_status( $comment_id, 'hold' );
 			}
 		}
-		
+
 		/*
 		 * Die() with or without screen based on JS availability
 		 */
 		private function cond_die( $message ) {
-			if ( isset( $_REQUEST['no_js'] ) && true == (boolean) $_REQUEST['no_js'] )
-				wp_die( __( $message ), "Safe Report Comments Notice", array('response' => 200 ) );
-			else
-				die( __( $message ) );
+			if ( isset( $_REQUEST['no_js'] ) && true == (boolean) $_REQUEST['no_js'] ) {
+				wp_die( esc_html( $message ), esc_html("Safe Report Comments Notice"), array('response' => 200 ) );
+			} else {
+				die( esc_html( $message ) );
+			}
 		}
-		
-		/* 
+
+		/*
 		 * Ajax callback to flag/report a comment
 		 */
-		public function flag_comment() {		
-			if ( (int) $_REQUEST[ 'comment_id' ] != $_REQUEST[ 'comment_id' ] || empty( $_REQUEST[ 'comment_id' ] ) )
+		public function flag_comment() {
+			if ( ! isset( $_REQUEST[ 'comment_id' ] ) || empty( $_REQUEST[ 'comment_id' ] ) || (int) $_REQUEST[ 'comment_id' ] != $_REQUEST[ 'comment_id' ] ) {
```

Comment:
No need for the `isset()` check if it's followed by `empty()` - `empty()` won't give any notice or warning if it's not set.